src/gui/options_imp.cpp

```diff
@@ -583,6 +587,13 @@ int options_imp::getProxyType() const
     }
 }
 
+int options_imp::limitStep(int value)
+{
+    static const int pow10[] = {1, 1, 10, 100, 1000, 10000, 10000};
+    int log = static_cast<int>(log10(value));
```

Comment:
I got rid of the lookup table to allow any potential big value in case we decide to allow limits above 100000 Kib/s in some future version. And I also check `value < 100` and simply return 1, which also covers negative values.